.github/workflows/ci.yaml

```
@@ -0,0 +1,209 @@
+        # passed; it is cumulative. Future versions of Stack might add behavior
+        # to --pedantic.
+        run : stack build --pedantic
+
+      - name: Test
+        run : stack test
+
+  # Cabal
+  cabal:
+    name: Cabal / GHC ${{ matrix.ghc }} ${{ matrix.project-variant }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        ghc:
+          - "8.0.2"
+          - "8.2.2"
+          - "8.4.4"
+          - "8.6.5"
+          - "8.8.4"
+          - "8.10.7"
+          - "9.0.2"
+          - "9.0.2"
+          - "9.2.8"
+          - "9.4.7"
+          - "9.6.3"
+        project-variant: [""]
+        include:
+          - ghc: 8.0.2
+            project-variant: -lower
+          - ghc: 9.8.1
+            project-variant: -upper
+
+      fail-fast: false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          show-progress: false
+
+      - name: Setup Haskell
+        uses: haskell-actions/setup@v2
+        id: setup-haskell
+        with:
+          ghc-version: ${{ matrix.ghc }}
+          cabal-update: false
+
+      - name: General Setup
+        run: |
+          cp .ci/cabal.project.local${{ matrix.project-variant}} \
+            cabal.project.local
+          # Print out some information for debugging purposes
+          ghc --version
+          cabal --version
+          cat cabal.project.local
+
+      - name: Setup CI
+        # This generates dist-newstyle/cache/plan.json with hashes for all
+        # dependencies, for cache invalidation.
+        run: |
+          cabal v2-update
+          cabal v2-build all --enable-tests --dry-run
+
+      - name: Restore cached dependencies
+        uses: actions/cache/restore@v3
+        id: cache
+        env:
+          key:
+            ${{ runner.os }}-ghc-${{ matrix.ghc }}-cabal-${{
+            steps.setup-haskell.outputs.cabal-version }}${{
+            matrix.project-variant }}
+        with:
+          path: ${{ steps.setup-haskell.outputs.cabal-store }}
+          key:
+            ${{ env.key }}-${{ hashFiles('cabal.project.local',
+            'dist-newstyle/cache/plan.json') }}
+          restore-keys: ${{ env.key }}-
+
+      - name: Install dependencies
+        run: cabal v2-build all --enable-tests --only-dependencies
+
+      # Cache dependencies already at this point, so that we do not have to
+      # rebuild them should the subsequent steps fail
+      - name: Save cached dependencies
+        uses: actions/cache/save@v4
+        # Trying to save over an existing cache gives distracting
+        # "Warning: Cache save failed." since they are immutable
+        if: steps.cache.outputs.cache-hit != 'true'
+        with:
+          path: ${{ steps.setup-haskell.outputs.cabal-store }}
+          key: ${{ steps.cache.outputs.cache-primary-key }}
+
+      - name: Build
+        run: cabal v2-build all
+
+      - name: Test
+        run: cabal v2-test --test-show-details=direct
```

```suggestion
        run: cabal v2-test --enable-tests --test-show-details=direct
```
Somehow only `-lower` was affected, but this does fix it.